demo/demo/tests/test_paddle.py

```diff
@@ -17,6 +20,141 @@
+    spreedly_api_url = payment_page.text.split("REACT_APP_SPREEDLY_API_URL: '")[-1].split("',")[0]
+
+    # Sanity checks on the variables
+    assert 'paddle' in checkout_api_url
+    checkout_api_url = f'{checkout_api_url}/checkout/{checkout_id}'
+    if not checkout_api_url.endswith('/'):
+        checkout_api_url += '/'
+
+    assert 'paddle' in ld_proxy_domain
+
+    assert 'spreedly' in spreedly_api_url
+    assert spreedly_api_url.endswith('v1'), 'Unexpected version of the spreedly API, this test might not work correctly'
+    if not spreedly_api_url.endswith('/'):
+        spreedly_api_url += '/'
+
+    # prepare the customer info request
+    customer_info_url = parse.urljoin(checkout_api_url, 'customer-info')
+    customer_info = requests.post(
+        customer_info_url,
+        json={
+            'data': {
+                'email': email,
+                'audience_optin': False,
+                'country_code': 'PL',
+                'postcode': 12345
+            }
+        }
+    )
+    payment_methods = customer_info.json().get(
+        'data', {}
+    ).get(
+        'available_payment_methods', []
+    )
+
+    # fetch the CARD payment method
+    for payment_method in payment_methods:
+        if payment_method['type'] == 'CARD':
+            assert 'spreedly_options' in payment_method
+            assert 'spreedly_environment_key' in payment_method['spreedly_options']
+            break
+    else:
+        assert False, 'Unexpected response from Paddle, payment method not found'
+
+    # Now we should have received the spreedly environment token.
+    spreedly_environment_key = payment_method['spreedly_options']['spreedly_environment_key']
+    assert spreedly_environment_key
+
+    # Set the payment method to CARD
+    set_payment_method_url = parse.urljoin(checkout_api_url, 'payment-method')
+    requests.patch(
+        set_payment_method_url,
+        json={
+            'data': {
+                "payment_method_type": "CARD"
+            }
+        }
+    )
+
+    # Send the card details to spreedly adn fetch the transaction token
+    spreedly_url = parse.urljoin(spreedly_api_url, f'payment_methods.json?environment_key={spreedly_environment_key}')
+    expire_date = date.today() + timedelta(90)
+    payment_response = session.post(spreedly_url, json={
+        "payment_method": {
+            "allow_blank_name": False,
+            "eligible_for_card_updater": False,
+            "credit_card": {
+                "country": "PL",
+                "email": email,
+                "full_name": "Test Example",
+                "kind": "credit_card",
+                "month": expire_date.strftime('%m'),
+                "number": card,
+                "verification_value": "123",
+                "year": expire_date.strftime('%Y'),
+                "zip": "12345"
+            }
+        }
+    })
+    transaction_token = payment_response.json().get('transaction', {}).get('payment_method', {}).get('token')
```

Comment:
Again, you probably don't want to use `get` here, you don't need defaults

Reply:
:+1: